be/src/olap/task/build_inverted_index.cpp

```diff
@@ -0,0 +1,461 @@
+                         << "[column=" << column_name << " referenced_column=" << column_idx
+                         << "]";
+            return Status::Error<ErrorCode::CE_CMD_PARAMS_ERROR>();
+        }
+        auto column = _cur_tablet_schema->column(column_idx);
+        _alter_column_uids.emplace_back(column.unique_id());
+    }
+    return Status::OK();
+}
+
+Status BuildInvertedIndex::update_inverted_index_info() {
+    // just do link files
+    LOG(INFO) << "begin to update_inverted_index_info, tablet=" << _tablet->tablet_id()
+            << ", is_drop_op=" << _is_drop_op;
+    for (auto i = 0; i < _input_rowsets.size(); ++i) {
+        auto input_rowset = _input_rowsets[i];
+        TabletSchemaSPtr output_rs_tablet_schema = std::make_shared<TabletSchema>();
+        output_rs_tablet_schema->copy_from(*input_rowset->tablet_schema());
+        if (_is_drop_op) {
+            output_rs_tablet_schema->update_indexes_from_thrift(_exist_indexes);
+        } else {
+            for (auto t_inverted_index : _alter_inverted_indexes) {
+                TabletIndex index;
+                index.init_from_thrift(t_inverted_index, *_cur_tablet_schema);
+                output_rs_tablet_schema->append_index(std::move(index));
+            }
+        }
+        // construct input rowset reader
+        RowsetReaderSharedPtr input_rs_reader;
+        RETURN_IF_ERROR(input_rowset->create_reader(&input_rs_reader));
+        
+        // construct output rowset writer
+        std::unique_ptr<RowsetWriter> output_rs_writer;
+        RowsetWriterContext context;
+        context.version = input_rs_reader->version();
+        context.rowset_state = VISIBLE;
+        context.segments_overlap = input_rs_reader->rowset()->rowset_meta()->segments_overlap();
+        context.tablet_schema = output_rs_tablet_schema;
+        context.newest_write_timestamp = input_rs_reader->newest_write_timestamp();
+        context.fs = input_rs_reader->rowset()->rowset_meta()->fs();
+        Status status = _tablet->create_rowset_writer(context, &output_rs_writer);
+        if (!status.ok()) {
+            return Status::Error<ErrorCode::ROWSET_BUILDER_INIT>();
+        }
+        std::set<int32_t> alter_column_ids_set(_alter_column_uids.begin(), _alter_column_uids.end());
+        RETURN_IF_ERROR(input_rowset->link_files_without_index_to(
+                        _tablet->tablet_path(), output_rs_writer->rowset_id(), alter_column_ids_set)); // build output rowset
+        
+        auto input_rowset_meta = input_rowset->rowset_meta();
+        RowsetMetaSharedPtr rowset_meta = std::make_shared<RowsetMeta>();
+        rowset_meta->set_num_rows(input_rowset_meta->num_rows());
+        rowset_meta->set_total_disk_size(input_rowset_meta->total_disk_size());
+        rowset_meta->set_data_disk_size(input_rowset_meta->data_disk_size());
+        rowset_meta->set_index_disk_size(input_rowset_meta->index_disk_size());
```

Comment:
index_disk_size should be updated after add/drop index

Reply:
this index size not contain inverted index size